<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exposure data • TwoSampleMR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Exposure data">
<meta property="og:description" content="TwoSampleMR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mrcieu/TwoSampleMR">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="exposure_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Exposure data</h1>
                        <h4 class="author">Gibran Hemani</h4>
            
      
      
      <div class="hidden name"><code>exposure.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>A data frame of the instruments for an exposure is required. Each line has the information for one variant for one exposure. The minimum information required for MR analysis is the following:</p>
<ul>
<li>
<code>SNP</code> - rs ID</li>
<li>
<code>beta</code> - The effect size. If the trait is binary then log(OR) should be used</li>
<li>
<code>se</code> - The standard error of the effect size</li>
<li>
<code>effect_allele</code> - The allele of the SNP which has the effect marked in <code>beta</code>
</li>
</ul>
<p>Other information that is useful for MR can also be provided:</p>
<ul>
<li>
<code>other_allele</code> - The non-effect allele</li>
<li>
<code>eaf</code> - The effect allele frequency</li>
<li>
<code>Phenotype</code> - The name of the phenotype for which the SNP has an effect</li>
</ul>
<p>You can also provide the following extra information:</p>
<ul>
<li>
<code>chr</code> - Physical position of variant (chromosome)</li>
<li>
<code>position</code> - Physical position of variant (position)</li>
<li>
<code>samplesize</code> - Sample size for estimating the effect size</li>
<li>
<code>ncase</code> - Number of cases</li>
<li>
<code>ncontrol</code> - Number of controls</li>
<li>
<code>pval</code> - The P-value for the SNP’s association with the exposure</li>
<li>
<code>units</code> - The units in which the effects are presented</li>
<li>
<code>gene</code> - The gene or other annotation for the the SNP</li>
</ul>
</div>
<div id="reading-in-from-a-file" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-in-from-a-file" class="anchor"></a>Reading in from a file</h2>
<p>The data can be read in from a text file using the <code>read_exposure_data</code> function. The file must have a header with column names corresponding to the columns described above.</p>
<div id="example-1-the-default-column-names-are-used" class="section level3">
<h3 class="hasAnchor">
<a href="#example-1-the-default-column-names-are-used" class="anchor"></a>Example 1: The default column names are used</h3>
<p>An example of a text file with the default column names is provided as part of the package, the first few rows look like this:</p>
<pre><code>Phenotype SNP beta se effect_allele other_allele eaf pval units gene samplesize
BMI rs10767664 0.19 0.0306122448979592 A T 0.78 5e-26 kg/m2 BDNF 225238
BMI rs13078807 0.1 0.0204081632653061 G A 0.2 4e-11 kg/m2 CADM2 221431
BMI rs1514175 0.07 0.0204081632653061 A G 0.43 8e-14 kg/m2 TNNI3K 207641
BMI rs1558902 0.39 0.0204081632653061 A T 0.42 5e-120 kg/m2 FTO 222476
BMI rs10968576 0.11 0.0204081632653061 G A 0.31 3e-13 kg/m2 LRRN6C 247166
BMI rs2241423 0.13 0.0204081632653061 G A 0.78 1e-18 kg/m2 LBXCOR1 227886</code></pre>
<p>The exact path to the file will be different on everyone’s computer, but it can be located like this:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bmi.txt"</span>, package<span class="op">=</span><span class="st">"TwoSampleMR"</span><span class="op">)</span></code></pre></div>
<p>You can read the data in like this:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_exposure_data.html">read_exposure_data</a></span><span class="op">(</span><span class="va">bmi_file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">bmi_exp_dat</span><span class="op">)</span>
<span class="co">#&gt;          SNP beta.exposure se.exposure effect_allele.exposure</span>
<span class="co">#&gt; 1 rs10767664          0.19  0.03061224                      A</span>
<span class="co">#&gt; 2 rs13078807          0.10  0.02040816                      G</span>
<span class="co">#&gt; 3  rs1514175          0.07  0.02040816                      A</span>
<span class="co">#&gt; 4  rs1558902          0.39  0.02040816                      A</span>
<span class="co">#&gt; 5 rs10968576          0.11  0.02040816                      G</span>
<span class="co">#&gt; 6  rs2241423          0.13  0.02040816                      G</span>
<span class="co">#&gt;   other_allele.exposure eaf.exposure pval.exposure units.exposure gene.exposure</span>
<span class="co">#&gt; 1                     T         0.78         5e-26          kg/m2          BDNF</span>
<span class="co">#&gt; 2                     A         0.20         4e-11          kg/m2         CADM2</span>
<span class="co">#&gt; 3                     G         0.43         8e-14          kg/m2        TNNI3K</span>
<span class="co">#&gt; 4                     T         0.42        5e-120          kg/m2           FTO</span>
<span class="co">#&gt; 5                     A         0.31         3e-13          kg/m2        LRRN6C</span>
<span class="co">#&gt; 6                     A         0.78         1e-18          kg/m2       LBXCOR1</span>
<span class="co">#&gt;   samplesize.exposure exposure mr_keep.exposure pval_origin.exposure</span>
<span class="co">#&gt; 1              225238      BMI             TRUE             reported</span>
<span class="co">#&gt; 2              221431      BMI             TRUE             reported</span>
<span class="co">#&gt; 3              207641      BMI             TRUE             reported</span>
<span class="co">#&gt; 4              222476      BMI             TRUE             reported</span>
<span class="co">#&gt; 5              247166      BMI             TRUE             reported</span>
<span class="co">#&gt; 6              227886      BMI             TRUE             reported</span>
<span class="co">#&gt;   units.exposure_dat id.exposure data_source.exposure</span>
<span class="co">#&gt; 1              kg/m2      mpnXLK             textfile</span>
<span class="co">#&gt; 2              kg/m2      mpnXLK             textfile</span>
<span class="co">#&gt; 3              kg/m2      mpnXLK             textfile</span>
<span class="co">#&gt; 4              kg/m2      mpnXLK             textfile</span>
<span class="co">#&gt; 5              kg/m2      mpnXLK             textfile</span>
<span class="co">#&gt; 6              kg/m2      mpnXLK             textfile</span></code></pre></div>
<p>The output from this function is a new data frame with standardised column names:</p>
<ul>
<li><code>SNP</code></li>
<li><code>exposure</code></li>
<li><code>beta.exposure</code></li>
<li><code>se.exposure</code></li>
<li><code>effect_allele.exposure</code></li>
<li><code>other_allele.exposure</code></li>
<li><code>eaf.exposure</code></li>
<li><code>mr_keep.exposure</code></li>
<li><code>pval.exposure</code></li>
<li><code>pval_origin.exposure</code></li>
<li><code>id.exposure</code></li>
<li><code>data_source.exposure</code></li>
<li><code>units.exposure</code></li>
<li><code>gene.exposure</code></li>
<li><code>samplesize.exposure</code></li>
</ul>
<p>The function attempts to match the columns to the ones it expects. It also checks that the data type is as expected.</p>
<p>If the required data for MR to be performed is not present (SNP name, effect size, standard error, effect allele) for a particular SNP, then the column <code>mr_keep.exposure</code> will be “FALSE.”</p>
</div>
<div id="example-2-the-text-file-has-non-default-column-names" class="section level3">
<h3 class="hasAnchor">
<a href="#example-2-the-text-file-has-non-default-column-names" class="anchor"></a>Example 2: The text file has non-default column names</h3>
<p>If the text file does not have default column names, this can still be read in as follows. Here are the first few rows of an example:</p>
<pre><code>rsid,effect,SE,a1,a2,a1_freq,p-value,Units,Gene,n
rs10767664,0.19,0.030612245,A,T,0.78,5.00E-26,kg/m2,BDNF,225238
rs13078807,0.1,0.020408163,G,A,0.2,4.00E-11,kg/m2,CADM2,221431
rs1514175,0.07,0.020408163,A,G,0.43,8.00E-14,kg/m2,TNNI3K,207641
rs1558902,0.39,0.020408163,A,T,0.42,5.00E-120,kg/m2,FTO,222476</code></pre>
<p>Note that this is a CSV file, with commas separating fields. The file is located here:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi2_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/bmi.csv"</span>, package<span class="op">=</span><span class="st">"TwoSampleMR"</span><span class="op">)</span></code></pre></div>
<p>To read in this data:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_exposure_data.html">read_exposure_data</a></span><span class="op">(</span>
    filename <span class="op">=</span> <span class="va">bmi2_file</span>,
    sep <span class="op">=</span> <span class="st">","</span>,
    snp_col <span class="op">=</span> <span class="st">"rsid"</span>,
    beta_col <span class="op">=</span> <span class="st">"effect"</span>,
    se_col <span class="op">=</span> <span class="st">"SE"</span>,
    effect_allele_col <span class="op">=</span> <span class="st">"a1"</span>,
    other_allele_col <span class="op">=</span> <span class="st">"a2"</span>,
    eaf_col <span class="op">=</span> <span class="st">"a1_freq"</span>,
    pval_col <span class="op">=</span> <span class="st">"p-value"</span>,
    units_col <span class="op">=</span> <span class="st">"Units"</span>,
    gene_col <span class="op">=</span> <span class="st">"Gene"</span>,
    samplesize_col <span class="op">=</span> <span class="st">"n"</span>
<span class="op">)</span>
<span class="co">#&gt; No phenotype name specified, defaulting to 'exposure'.</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">bmi_exp_dat</span><span class="op">)</span>
<span class="co">#&gt;          SNP beta.exposure se.exposure effect_allele.exposure</span>
<span class="co">#&gt; 1 rs10767664          0.19  0.03061224                      A</span>
<span class="co">#&gt; 2 rs13078807          0.10  0.02040816                      G</span>
<span class="co">#&gt; 3  rs1514175          0.07  0.02040816                      A</span>
<span class="co">#&gt; 4  rs1558902          0.39  0.02040816                      A</span>
<span class="co">#&gt; 5 rs10968576          0.11  0.02040816                      G</span>
<span class="co">#&gt; 6  rs2241423          0.13  0.02040816                      G</span>
<span class="co">#&gt;   other_allele.exposure eaf.exposure pval.exposure units.exposure gene.exposure</span>
<span class="co">#&gt; 1                     T         0.78         5e-26          kg/m2          BDNF</span>
<span class="co">#&gt; 2                     A         0.20         4e-11          kg/m2         CADM2</span>
<span class="co">#&gt; 3                     G         0.43         8e-14          kg/m2        TNNI3K</span>
<span class="co">#&gt; 4                     T         0.42        5e-120          kg/m2           FTO</span>
<span class="co">#&gt; 5                     A         0.31         3e-13          kg/m2        LRRN6C</span>
<span class="co">#&gt; 6                     A         0.78         1e-18          kg/m2       LBXCOR1</span>
<span class="co">#&gt;   samplesize.exposure exposure mr_keep.exposure pval_origin.exposure</span>
<span class="co">#&gt; 1              225238 exposure             TRUE             reported</span>
<span class="co">#&gt; 2              221431 exposure             TRUE             reported</span>
<span class="co">#&gt; 3              207641 exposure             TRUE             reported</span>
<span class="co">#&gt; 4              222476 exposure             TRUE             reported</span>
<span class="co">#&gt; 5              247166 exposure             TRUE             reported</span>
<span class="co">#&gt; 6              227886 exposure             TRUE             reported</span>
<span class="co">#&gt;   units.exposure_dat id.exposure data_source.exposure</span>
<span class="co">#&gt; 1              kg/m2      QdNytt             textfile</span>
<span class="co">#&gt; 2              kg/m2      QdNytt             textfile</span>
<span class="co">#&gt; 3              kg/m2      QdNytt             textfile</span>
<span class="co">#&gt; 4              kg/m2      QdNytt             textfile</span>
<span class="co">#&gt; 5              kg/m2      QdNytt             textfile</span>
<span class="co">#&gt; 6              kg/m2      QdNytt             textfile</span></code></pre></div>
<p>If the <code>Phenotype</code> column is not provided (as is the case in this example) then it will assume that the phenotype’s name is simply “exposure.” This is entered in the <code>exposure</code> column. It can be renamed manually:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_exp_dat</span><span class="op">$</span><span class="va">exposure</span> <span class="op">&lt;-</span> <span class="st">"BMI"</span></code></pre></div>
</div>
</div>
<div id="using-an-existing-data-frame" class="section level2">
<h2 class="hasAnchor">
<a href="#using-an-existing-data-frame" class="anchor"></a>Using an existing data frame</h2>
<p>If the data already exists as a data frame in R then it can be converted into the correct format using the <code>format_data</code> function. For example, here is some randomly created data:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">random_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    SNP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"rs1"</span>, <span class="st">"rs2"</span><span class="op">)</span>,
    beta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,
    se <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,
    effect_allele <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"T"</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">random_df</span>
<span class="co">#&gt;   SNP beta se effect_allele</span>
<span class="co">#&gt; 1 rs1    1  1             A</span>
<span class="co">#&gt; 2 rs2    2  2             T</span></code></pre></div>
<p>This can be formatted like so:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">random_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_data.html">format_data</a></span><span class="op">(</span><span class="va">random_df</span>, type<span class="op">=</span><span class="st">"exposure"</span><span class="op">)</span>
<span class="co">#&gt; No phenotype name specified, defaulting to 'exposure'.</span>
<span class="co">#&gt; Warning in format_data(random_df, type = "exposure"): The following columns are not present but are helpful for harmonisation</span>
<span class="co">#&gt; other_alleleeaf</span>
<span class="co">#&gt; Inferring p-values</span>
<span class="va">random_exp_dat</span>
<span class="co">#&gt;   SNP beta.exposure se.exposure effect_allele.exposure exposure</span>
<span class="co">#&gt; 1 rs1             1           1                      A exposure</span>
<span class="co">#&gt; 2 rs2             2           2                      T exposure</span>
<span class="co">#&gt;   mr_keep.exposure pval.exposure pval_origin.exposure id.exposure</span>
<span class="co">#&gt; 1             TRUE     0.1586553             inferred      n0RQkL</span>
<span class="co">#&gt; 2             TRUE     0.1586553             inferred      n0RQkL</span>
<span class="co">#&gt;   other_allele.exposure eaf.exposure</span>
<span class="co">#&gt; 1                    NA           NA</span>
<span class="co">#&gt; 2                    NA           NA</span></code></pre></div>
</div>
<div id="obtaining-instruments-from-existing-catalogues" class="section level2">
<h2 class="hasAnchor">
<a href="#obtaining-instruments-from-existing-catalogues" class="anchor"></a>Obtaining instruments from existing catalogues</h2>
<p>A number of sources of instruments have already been curated and are available for use. They are provided as data objects in the <code>MRInstruments</code> package. To install:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"MRCIEU/MRInstruments"</span><span class="op">)</span></code></pre></div>
<p>This package contains a number of data.frames, each of which is a repository of SNP-trait associations. How to access the data frames is detailed below:</p>
<div id="gwas-catalog" class="section level3">
<h3 class="hasAnchor">
<a href="#gwas-catalog" class="anchor"></a>GWAS catalog</h3>
<p>The NHGRI-EBI GWAS catalog contains a catalog of significant associations obtained from GWASs. This version of the data is filtered and harmonised to contain associations that have the required data to perform MR, to ensure that the units used to report effect sizes from a particular study are all the same, and other data cleaning operations.</p>
<p>To use the GWAS catalog:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MRCIEU/MRInstruments">MRInstruments</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gwas_catalog</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gwas_catalog</span><span class="op">)</span>
<span class="co">#&gt;                                                 Phenotype_simple</span>
<span class="co">#&gt; 1                           Eosinophil percentage of white cells</span>
<span class="co">#&gt; 2                                              Eosinophil counts</span>
<span class="co">#&gt; 3 Medication use (agents acting on the renin-angiotensin system)</span>
<span class="co">#&gt; 4                                       Post bronchodilator FEV1</span>
<span class="co">#&gt; 5                         DNA methylation variation (age effect)</span>
<span class="co">#&gt; 6                                         Ankylosing spondylitis</span>
<span class="co">#&gt;                                                MAPPED_TRAIT_EFO</span>
<span class="co">#&gt; 1                           eosinophil percentage of leukocytes</span>
<span class="co">#&gt; 2                                              eosinophil count</span>
<span class="co">#&gt; 3 Agents acting on the renin-angiotensin system use measurement</span>
<span class="co">#&gt; 4          forced expiratory volume, response to bronchodilator</span>
<span class="co">#&gt; 5                                               DNA methylation</span>
<span class="co">#&gt; 6                                        ankylosing spondylitis</span>
<span class="co">#&gt;                                                              MAPPED_TRAIT_EFO_URI</span>
<span class="co">#&gt; 1                                            http://www.ebi.ac.uk/efo/EFO_0007991</span>
<span class="co">#&gt; 2                                            http://www.ebi.ac.uk/efo/EFO_0004842</span>
<span class="co">#&gt; 3                                            http://www.ebi.ac.uk/efo/EFO_0009931</span>
<span class="co">#&gt; 4 http://www.ebi.ac.uk/efo/EFO_0004314, http://purl.obolibrary.org/obo/GO_0097366</span>
<span class="co">#&gt; 5                                       http://purl.obolibrary.org/obo/GO_0006306</span>
<span class="co">#&gt; 6                                            http://www.ebi.ac.uk/efo/EFO_0003898</span>
<span class="co">#&gt;                                                                                                                                                Initial_sample_description</span>
<span class="co">#&gt; 1                                                                                                                                   172,378 European ancestry individuals</span>
<span class="co">#&gt; 2                                                                                                                                   172,275 European ancestry individuals</span>
<span class="co">#&gt; 3                                                                                                      62,752 European ancestry cases, 174,778 European ancestry controls</span>
<span class="co">#&gt; 4 10,094 European ancestry current and former smoker individuals, 3,260 African American current and former smoker individuals, 178 current and former smoker individuals</span>
<span class="co">#&gt; 5                                                                                                                                                   Up to 954 individuals</span>
<span class="co">#&gt; 6                                                    921 Turkish ancestry cases, 907 Turkish ancestry controls, 422 Iranian ancestry cases, 754 Iranian ancestry controls</span>
<span class="co">#&gt;   Replication_sample_description STUDY.ACCESSION</span>
<span class="co">#&gt; 1                           &lt;NA&gt;      GCST004600</span>
<span class="co">#&gt; 2                           &lt;NA&gt;      GCST004606</span>
<span class="co">#&gt; 3                           &lt;NA&gt;      GCST007930</span>
<span class="co">#&gt; 4                           &lt;NA&gt;      GCST003262</span>
<span class="co">#&gt; 5                           &lt;NA&gt;      GCST006660</span>
<span class="co">#&gt; 6                           &lt;NA&gt;      GCST007844</span>
<span class="co">#&gt;                                                        Phenotype Phenotype_info</span>
<span class="co">#&gt; 1                           Eosinophil percentage of white cells               </span>
<span class="co">#&gt; 2                                              Eosinophil counts               </span>
<span class="co">#&gt; 3 Medication use (agents acting on the renin-angiotensin system)               </span>
<span class="co">#&gt; 4                                       Post bronchodilator FEV1               </span>
<span class="co">#&gt; 5                         DNA methylation variation (age effect)               </span>
<span class="co">#&gt; 6                                         Ankylosing spondylitis               </span>
<span class="co">#&gt;   PubmedID   Author Year        SNP chr bp_ens_GRCh38   Region       gene</span>
<span class="co">#&gt; 1 27863252 Astle WJ 2016  rs1000005  21      33060745 21q22.11 AP000282.2</span>
<span class="co">#&gt; 2 27863252 Astle WJ 2016  rs1000005  21      33060745 21q22.11 AP000282.2</span>
<span class="co">#&gt; 3 31015401     Wu Y 2019  rs1000010   3      11562645   3p25.3      VGLL4</span>
<span class="co">#&gt; 4 26634245  Lutz SM 2015 rs10000225   4     144312789  4q31.21 Intergenic</span>
<span class="co">#&gt; 5 30348214  Zhang Q 2018 rs10000513   4     160334994   4q32.1         NR</span>
<span class="co">#&gt; 6 30946743     Li Z 2019 rs10000518   4      11502867  4p15.33     HS3ST1</span>
<span class="co">#&gt;               Gene_ens effect_allele other_allele        beta          se  pval</span>
<span class="co">#&gt; 1 AP000282.2,LINC00945             C            G -0.02652552 0.003826531 2e-13</span>
<span class="co">#&gt; 2 AP000282.2,LINC00945             C            G -0.02481715 0.003571429 7e-12</span>
<span class="co">#&gt; 3                                  G            A -0.03724189 0.006377551 6e-09</span>
<span class="co">#&gt; 4           Intergenic             A            T -0.04400000 0.009420188 3e-06</span>
<span class="co">#&gt; 5                   NR          &lt;NA&gt;         &lt;NA&gt;          NA          NA 4e-08</span>
<span class="co">#&gt; 6                                  G            A  0.73396926          NA 6e-06</span>
<span class="co">#&gt;              units      eaf date_added_to_MRBASE</span>
<span class="co">#&gt; 1    unit decrease 0.589400           2019-08-29</span>
<span class="co">#&gt; 2    unit decrease 0.589400           2019-08-29</span>
<span class="co">#&gt; 3    unit decrease 0.351806           2019-08-29</span>
<span class="co">#&gt; 4 NR unit decrease 0.350000           2019-08-29</span>
<span class="co">#&gt; 5             &lt;NA&gt;       NA           2019-08-29</span>
<span class="co">#&gt; 6             &lt;NA&gt;       NA           2019-08-29</span></code></pre></div>
<p>For example, to obtain instruments for body mass index using the Speliotes et al 2010 study:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_gwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">gwas_catalog</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Speliotes"</span>, <span class="va">Author</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">Phenotype</span> <span class="op">==</span> <span class="st">"Body mass index"</span><span class="op">)</span>
<span class="va">bmi_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_data.html">format_data</a></span><span class="op">(</span><span class="va">bmi_gwas</span><span class="op">)</span></code></pre></div>
</div>
<div id="metabolites" class="section level3">
<h3 class="hasAnchor">
<a href="#metabolites" class="anchor"></a>Metabolites</h3>
<p>Independent top hits from GWASs on 121 metabolites in whole blood are stored in the <code>metab_qtls</code> data object. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/metab_qtls.html">?metab_qtls</a></code> to get more information.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">metab_qtls</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">metab_qtls</span><span class="op">)</span>
<span class="co">#&gt;   phenotype chromosome  position       SNP effect_allele other_allele      eaf</span>
<span class="co">#&gt; 1     AcAce          8   9181395 rs2169387             G            A 0.870251</span>
<span class="co">#&gt; 2     AcAce         11 116648917  rs964184             C            G 0.857715</span>
<span class="co">#&gt; 3       Ace          6  12042473 rs6933521             C            T 0.120256</span>
<span class="co">#&gt; 4       Ala          2  27730940 rs1260326             C            T 0.638817</span>
<span class="co">#&gt; 5       Ala          2  65220910 rs2160387             C            T 0.403170</span>
<span class="co">#&gt; 6       Ala         12  47201814 rs4554975             G            A 0.644059</span>
<span class="co">#&gt;        beta       se     pval n_studies     n</span>
<span class="co">#&gt; 1  0.085630 0.015451 3.61e-08        11 19257</span>
<span class="co">#&gt; 2 -0.096027 0.014624 6.71e-11        11 19261</span>
<span class="co">#&gt; 3 -0.091667 0.015885 8.10e-09        14 24742</span>
<span class="co">#&gt; 4 -0.104582 0.009940 7.40e-26        13 22569</span>
<span class="co">#&gt; 5 -0.071001 0.009603 1.49e-13        14 24793</span>
<span class="co">#&gt; 6 -0.069135 0.009598 6.12e-13        14 24792</span></code></pre></div>
<p>For example, to obtain instruments for the Alanine:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ala_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_metab_qtls.html">format_metab_qtls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">metab_qtls</span>, <span class="va">phenotype</span><span class="op">==</span><span class="st">"Ala"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="proteins" class="section level3">
<h3 class="hasAnchor">
<a href="#proteins" class="anchor"></a>Proteins</h3>
<p>Independent top hits from GWASs on 47 protein levels in whole blood are stored in the <code>proteomic_qtls</code> data object. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/proteomic_qtls.html">?proteomic_qtls</a></code> to get more information.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">proteomic_qtls</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">proteomic_qtls</span><span class="op">)</span>
<span class="co">#&gt;   analyte chr  position        SNP  gene location annotation other_allele</span>
<span class="co">#&gt; 1   CFHR1   1 196698945 rs12144939   CFH      cis   missense            T</span>
<span class="co">#&gt; 2    IL6r   1 154425456 rs12126142  IL6R      cis   missense            A</span>
<span class="co">#&gt; 3   ApoA4  11 116677723  rs1263167 APOA4      cis intergenic            G</span>
<span class="co">#&gt; 4    SELE   9 136149399   rs507666   ABO    trans   intronic            A</span>
<span class="co">#&gt; 5 FetuinA   3 186335941  rs2070633  AHSG      cis   missense            T</span>
<span class="co">#&gt; 6     ACE  17  61566031     rs4343   ACE      cis synonymous            A</span>
<span class="co">#&gt;   effect_allele   eaf   maf      pval   beta         se</span>
<span class="co">#&gt; 1             G 0.643 0.357 8.99e-143 -1.108 0.04355258</span>
<span class="co">#&gt; 2             G 0.608 0.392 1.81e-106  0.850 0.03878364</span>
<span class="co">#&gt; 3             A 0.803 0.197  2.64e-54 -0.919 0.05922332</span>
<span class="co">#&gt; 4             G 0.809 0.191  1.01e-52 -0.882 0.05771545</span>
<span class="co">#&gt; 5             C 0.676 0.324  2.88e-44 -0.629 0.04506925</span>
<span class="co">#&gt; 6             G 0.508 0.492  6.66e-44  0.493 0.03547679</span></code></pre></div>
<p>For example, to obtain instruments for the ApoH protein:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">apoh_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_proteomic_qtls.html">format_proteomic_qtls</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">proteomic_qtls</span>, <span class="va">analyte</span><span class="op">==</span><span class="st">"ApoH"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in format_data(proteomic_qtls_subset, type = type, phenotype_col =</span>
<span class="co">#&gt; "analyte"): effect_allele column is not character data. Coercing...</span>
<span class="co">#&gt; Warning in format_data(proteomic_qtls_subset, type = type, phenotype_col =</span>
<span class="co">#&gt; "analyte"): other_allele column is not character data. Coercing...</span></code></pre></div>
</div>
<div id="gene-expression-levels" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-expression-levels" class="anchor"></a>Gene expression levels</h3>
<p>Independent top hits from GWASs on 32432 gene identifiers and in 44 tissues are available from the GTEX study in <code>gtex_eqtl</code>. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/gtex_eqtl.html">?gtex_eqtl</a></code> to get more information.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gtex_eqtl</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gtex_eqtl</span><span class="op">)</span>
<span class="co">#&gt;                 tissue     gene_name gene_start         SNP snp_position</span>
<span class="co">#&gt; 1 Adipose Subcutaneous RP4-669L17.10   1:317720   rs2519065     1:787151</span>
<span class="co">#&gt; 2 Adipose Subcutaneous RP11-206L10.1   1:661611  rs11804171     1:723819</span>
<span class="co">#&gt; 3 Adipose Subcutaneous RP11-206L10.3   1:677193 rs149110718     1:759227</span>
<span class="co">#&gt; 4 Adipose Subcutaneous RP11-206L10.2   1:700306 rs148649543     1:752796</span>
<span class="co">#&gt; 5 Adipose Subcutaneous RP11-206L10.9   1:714150  rs12184279     1:717485</span>
<span class="co">#&gt; 6 Adipose Subcutaneous RP11-206L10.8   1:736259  rs10454454     1:754954</span>
<span class="co">#&gt;   effect_allele other_allele      beta        se        pval   n</span>
<span class="co">#&gt; 1             A            G  0.551788 0.0747180 2.14627e-12 298</span>
<span class="co">#&gt; 2             A            T -0.917475 0.1150060 4.99967e-14 298</span>
<span class="co">#&gt; 3             T            C  0.807571 0.1776530 8.44694e-06 298</span>
<span class="co">#&gt; 4             T            C  0.745393 0.0958531 1.82660e-13 298</span>
<span class="co">#&gt; 5             A            C  1.927250 0.2247390 9.55098e-16 298</span>
<span class="co">#&gt; 6             A            G  1.000400 0.1787470 5.61079e-08 298</span></code></pre></div>
<p>For example, to obtain instruments for the IRAK1BP1 gene expression levels in subcutaneous adipose tissue:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">irak1bp1_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_gtex_eqtl.html">format_gtex_eqtl</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">gtex_eqtl</span>, <span class="va">gene_name</span><span class="op">==</span><span class="st">"IRAK1BP1"</span> <span class="op">&amp;</span> <span class="va">tissue</span> <span class="op">==</span> <span class="st">"Adipose Subcutaneous"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in format_data(gtex_eqtl_subset, type = type, phenotype_col = type, : The following columns are not present but are helpful for harmonisation</span>
<span class="co">#&gt; eaf</span>
<span class="co">#&gt; Inferring p-values</span></code></pre></div>
</div>
<div id="dna-methylation-levels" class="section level3">
<h3 class="hasAnchor">
<a href="#dna-methylation-levels" class="anchor"></a>DNA methylation levels</h3>
<p>Independent top hits from GWASs on 0 DNA methylation levels in whole blood across 5 time points are available from the ARIES study in <code>aries_mqtl</code>. Use <code><a href="https://rdrr.io/pkg/MRInstruments/man/aries_mqtl.html">?aries_mqtl</a></code> to get more information.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">aries_mqtl</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">aries_mqtl</span><span class="op">)</span>
<span class="co">#&gt;          SNP timepoint        cpg    beta        pval         se snp_chr</span>
<span class="co">#&gt; 1 esv2656832         1 cg21826606  0.3459 1.60408e-26 0.03265336       1</span>
<span class="co">#&gt; 2 esv2658098         1 cg22681495 -0.6263 1.55765e-66 0.03643240      15</span>
<span class="co">#&gt; 3 esv2660043         1 cg24276624 -0.5772 3.16370e-26 0.05481823      11</span>
<span class="co">#&gt; 4 esv2660043         1 cg11157765 -0.5423 1.33928e-22 0.05583777      11</span>
<span class="co">#&gt; 5 esv2660673         1 cg05832925 -0.5919 2.88011e-50 0.03982467      11</span>
<span class="co">#&gt; 6 esv2660769         1 cg05859533 -0.6224 1.49085e-58 0.03868158      16</span>
<span class="co">#&gt;    snp_pos effect_allele other_allele    eaf   sex   age    units</span>
<span class="co">#&gt; 1 25591901             I            R 0.3974 mixed Birth SD units</span>
<span class="co">#&gt; 2 86057007             D            R 0.2076 mixed Birth SD units</span>
<span class="co">#&gt; 3 69982552             D            R 0.1450 mixed Birth SD units</span>
<span class="co">#&gt; 4 69982552             D            R 0.1450 mixed Birth SD units</span>
<span class="co">#&gt; 5 74024905             D            R 0.1671 mixed Birth SD units</span>
<span class="co">#&gt; 6 57725395             D            R 0.2136 mixed Birth SD units</span>
<span class="co">#&gt;   island_location cpg_chr  cpg_pos    gene gene_location cis_trans</span>
<span class="co">#&gt; 1         N_Shore       1 25593055                             cis</span>
<span class="co">#&gt; 2                      15 86058755  AKAP13          Body       cis</span>
<span class="co">#&gt; 3                      11 69982941    ANO1          Body       cis</span>
<span class="co">#&gt; 4                      11 69982996    ANO1          Body       cis</span>
<span class="co">#&gt; 5         S_Shelf      11 74026371                             cis</span>
<span class="co">#&gt; 6                      16 57727230 CCDC135       TSS1500       cis</span></code></pre></div>
<p>For example, to obtain instruments for cg25212131 CpG DNA methylation levels in at birth:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cg25212131_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_aries_mqtl.html">format_aries_mqtl</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">aries_mqtl</span>, <span class="va">cpg</span><span class="op">==</span><span class="st">"cg25212131"</span> <span class="op">&amp;</span> <span class="va">age</span> <span class="op">==</span> <span class="st">"Birth"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="ieu-gwas-database" class="section level3">
<h3 class="hasAnchor">
<a href="#ieu-gwas-database" class="anchor"></a>IEU GWAS database</h3>
<p>The IEU GWAS database contains the entire summary statistics for thousands of GWASs. You can browse them here: <a href="https://gwas.mrcieu.ac.uk/" class="uri">https://gwas.mrcieu.ac.uk/</a></p>
<p>You can use this database to define the instruments for a particular exposure. You can also use this database to obtain the effects for constructing polygenic risk scores using different p-value thresholds.</p>
<p>You can check the status of the API:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ieugwasr/man/api_status.html">api_status</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="co">#&gt; $`API version`</span>
<span class="co">#&gt; [1] "3.6.16"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Access</span>
<span class="co">#&gt; [1] "public"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Neo4j status`</span>
<span class="co">#&gt; [1] "Available"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`ElasticSearch status`</span>
<span class="co">#&gt; [1] "Available"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`LD reference panel`</span>
<span class="co">#&gt; [1] "Available"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`1000 genomes annotation VCF`</span>
<span class="co">#&gt; [1] "Available"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`PLINK executable`</span>
<span class="co">#&gt; [1] "Available"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Cromwell</span>
<span class="co">#&gt; [1] "Available"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Total associations`</span>
<span class="co">#&gt; [1] 1.84716e+11</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Total complete datasets`</span>
<span class="co">#&gt; [1] 39719</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Total public datasets`</span>
<span class="co">#&gt; [1] 39603</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "ApiStatus"</span></code></pre></div>
<p>To obtain a list and details about the available GWASs do the following:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_outcomes.html">available_outcomes</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ao</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 22</span>
<span class="co">#&gt;   id      trait    note  group_name  year    mr author consortium sex   priority</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;      &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;    &lt;int&gt;</span>
<span class="co">#&gt; 1 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 2 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 3 ukb-e-~ Symptom~ NA    public      2020     1 Pan-U~ NA         Male~        0</span>
<span class="co">#&gt; 4 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 5 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 6 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; # ... with 12 more variables: population &lt;chr&gt;, unit &lt;chr&gt;, sample_size &lt;int&gt;,</span>
<span class="co">#&gt; #   nsnp &lt;int&gt;, build &lt;chr&gt;, subcategory &lt;chr&gt;, category &lt;chr&gt;, ontology &lt;chr&gt;,</span>
<span class="co">#&gt; #   ncase &lt;int&gt;, ncontrol &lt;int&gt;, pmid &lt;int&gt;, sd &lt;dbl&gt;</span></code></pre></div>
<p>For information about authentication see <a href="https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication">https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication</a>.</p>
<p>The <code>available_outcomes</code> function returns a table of all the available studies in the database. Each study has a unique ID. e.g.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">ao</span>, select<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">trait</span>, <span class="va">id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   trait                               id                    </span>
<span class="co">#&gt;   &lt;chr&gt;                               &lt;chr&gt;                 </span>
<span class="co">#&gt; 1 ENSG00000198715                     eqtl-a-ENSG00000198715</span>
<span class="co">#&gt; 2 ENSG00000179950                     eqtl-a-ENSG00000179950</span>
<span class="co">#&gt; 3 Symptoms involving digestive system ukb-e-561_CSA         </span>
<span class="co">#&gt; 4 ENSG00000106683                     eqtl-a-ENSG00000106683</span>
<span class="co">#&gt; 5 ENSG00000143819                     eqtl-a-ENSG00000143819</span>
<span class="co">#&gt; 6 ENSG00000139574                     eqtl-a-ENSG00000139574</span></code></pre></div>
<p>To extract instruments for a particular trait using a particular study, for example to obtain SNPs for body mass index using the Locke et al 2015 GIANT study, you specify the study ID as follows:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi2014_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_instruments.html">extract_instruments</a></span><span class="op">(</span>outcomes<span class="op">=</span><span class="st">'ieu-a-2'</span><span class="op">)</span></code></pre></div>
<p>This returns a set of LD clumped SNPs that are GWAS significant for BMI. You can specify various parameters for this function:</p>
<ul>
<li>
<code>p1</code> = P-value threshold for keeping a SNP</li>
<li>
<code>clump</code> = Whether or not to return independent SNPs only (default=TRUE)</li>
<li>
<code>r2</code> = The maximum LD R-square allowed between returned SNPs</li>
<li>
<code>kb</code> = The distance in which to search for LD R-square values</li>
</ul>
<p>By changing changing the <code>p1</code> parameter it is possible to obtain SNP effects for constructing polygenic risk scores.</p>
</div>
</div>
<div id="clumping" class="section level2">
<h2 class="hasAnchor">
<a href="#clumping" class="anchor"></a>Clumping</h2>
<p>For standard two sample MR it is important to ensure that the instruments for the exposure are independent. Once instruments have been identified for an exposure variable, the IEU GWAS database can be used to perform clumping.</p>
<p>You can provide a list of SNP IDs, the SNPs will be extracted from 1000 genomes data, LD calculated between them, and amongst those SNPs that have LD R-square above the specified threshold only the SNP with the lowest P-value will be retained. To do this, use the following command:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clump_data.html">clump_data</a></span><span class="op">(</span><span class="va">bmi_exp_dat</span><span class="op">)</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Clumping QdNytt, 30 variants, using EUR population reference</span>
<span class="co">#&gt; Removing 3 of 30 variants due to LD with other variants or absence from LD reference panel</span></code></pre></div>
<p>The <code>clump_data</code> command takes any data frame that has been formatted to be an exposure data type of data frame. Note that for the instruments in the R/MRInstruments package the SNPs are already LD clumped.</p>
<p><strong>Note:</strong> The LD reference panel only includes SNPs (no INDELs). There are five super-populations from which LD can be calculated, by default European samples are used. Only SNPs with MAF &gt; 0.01 within-population are available.</p>
<p><strong>NOTE:</strong> If a variant is dropped from your unclumped data it could be because it is absent from the reference panel. For more flexibility, including using your own LD reference data, see here: <a href="https://mrcieu.github.io/ieugwasr/" class="uri">https://mrcieu.github.io/ieugwasr/</a></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
