<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Harmonise data • TwoSampleMR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Harmonise data">
<meta property="og:description" content="TwoSampleMR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mrcieu/TwoSampleMR">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="harmonise_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Harmonise data</h1>
                        <h4 class="author">Gibran Hemani</h4>
            
      
      
      <div class="hidden name"><code>harmonise.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The exposure data and outcome data are now obtained, e.g.:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_outcomes.html">available_outcomes</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="va">bmi_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_instruments.html">extract_instruments</a></span><span class="op">(</span>outcomes<span class="op">=</span><span class="st">'ieu-a-2'</span><span class="op">)</span>
<span class="va">chd_out_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span><span class="op">(</span>snps <span class="op">=</span> <span class="va">bmi_exp_dat</span><span class="op">$</span><span class="va">SNP</span>, outcomes <span class="op">=</span> <span class="st">'ieu-a-7'</span><span class="op">)</span>
<span class="co">#&gt; Extracting data for 79 SNP(s) from 1 GWAS(s)</span></code></pre></div>
<p>but it is important to harmonise the effects. This means that the effect of a SNP on the exposure and the effect of that SNP on the outcome must each correspond to the same allele.</p>
<p><strong>Note:</strong> The IEU GWAS database contains data that is already harmonised, meaning that the non-effect allele is aligned to the human genome reference sequence (build 37). It’s still recommended to harmonise, but in principle everything should be on the forward strand and effect alleles always relating to the same allele. Some discrepancies could arise if there are multi-allelic variants that are represented as different bi-allelic variants in different studies.</p>
<p>To harmonise the exposure and outcome data, do the following:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/harmonise_data.html">harmonise_data</a></span><span class="op">(</span>
    exposure_dat <span class="op">=</span> <span class="va">bmi_exp_dat</span>, 
    outcome_dat <span class="op">=</span> <span class="va">chd_out_dat</span>
<span class="op">)</span>
<span class="co">#&gt; Harmonising Body mass index || id:ieu-a-2 (ieu-a-2) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span></code></pre></div>
<p>This creates a new data frame that has the exposure data and outcome data combined.</p>
<p>If there were 3 exposure traits and 3 outcome traits then there will be 9 sets of harmonisations being performed - harmonising the SNP effects of exposure trait 1 against outcome trait 1; exposure trait 1 against outcome trait 2; and so on.</p>
</div>
<div id="dealing-with-strand-issues" class="section level2">
<h2 class="hasAnchor">
<a href="#dealing-with-strand-issues" class="anchor"></a>Dealing with strand issues</h2>
<p>Recent GWASs typically present the effects of a SNP in reference to the allele on the forward strand. But as reference panels are updated the forward strand sometimes changes, and GWASs from a few years ago aren’t guaranteed to be using forward strand conventions.</p>
<p>Some examples are shown below:</p>
<div id="correct-unambigious" class="section level3">
<h3 class="hasAnchor">
<a href="#correct-unambigious" class="anchor"></a>Correct, unambigious</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = G

outcome effect = 0.05
effect allele = A
other allele = G</code></pre>
<p>Here the effect allele on the exposure and the outcome is the same</p>
</div>
<div id="incorrect-reference-unambigious" class="section level3">
<h3 class="hasAnchor">
<a href="#incorrect-reference-unambigious" class="anchor"></a>Incorrect reference, unambigious</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = G

outcome effect = -0.05
effect allele = C
other allele = T</code></pre>
<p>Here the outcome GWAS is presenting the effect for the alternate allele on the reverse strand. We need to flip the outcome effect to 0.05 to correspond to the same allele as the exposure GWAS on the forward strand.</p>
</div>
<div id="ambiguous" class="section level3">
<h3 class="hasAnchor">
<a href="#ambiguous" class="anchor"></a>Ambiguous</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = G

outcome effect = -0.05
effect allele = A
other allele = C</code></pre>
<p>Here the alleles do not correspond for the same SNP, so this SNP will be discarded from the analysis.</p>
</div>
<div id="palindromic-snp-inferrable" class="section level3">
<h3 class="hasAnchor">
<a href="#palindromic-snp-inferrable" class="anchor"></a>Palindromic SNP, inferrable</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = T
effect allele frequency = 0.11

outcome effect = -0.05
effect allele = A
other allele = T
effect allele frequency = 0.91</code></pre>
<p>Here the alleles correspond, but it is a palindromic SNP, such that the alleles on the forward strand are the same as on the reverse strand (A/T on forward is T/A on the reverse). However, the allele frequency of the effect allele gives us information - if the outcome effect allele (A) were on the forward strand we would expect it to have a low allele frequency, but given it has a high frequency (0.91) we infer that the outcome GWAS is presenting the effect on the reverse strand for the alternative allele. We would flip the effect to 0.05 for the outcome GWAS.</p>
</div>
<div id="palindromic-snp-not-inferrable" class="section level3">
<h3 class="hasAnchor">
<a href="#palindromic-snp-not-inferrable" class="anchor"></a>Palindromic SNP, not inferrable</h3>
<pre><code>exposure effect = 0.5
effect allele = A
other allele = T
effect allele frequency = 0.50

outcome effect = -0.05
effect allele = A
other allele = T
effect allele frequency = 0.50</code></pre>
<p>This is similar to the above, except the allele frequency no longer gives us information about the strand. We would discard this SNP. This is done for any palindromic SNPs that have minor allele frequency above 0.42.</p>
</div>
<div id="options" class="section level3">
<h3 class="hasAnchor">
<a href="#options" class="anchor"></a>Options</h3>
<p>There are three options to harmonising the data.</p>
<ol style="list-style-type: decimal">
<li>Assume all alleles are presented on the forward strand</li>
<li>Try to infer the forward strand alleles using allele frequency information</li>
<li>Correct the strand for non-palindromic SNPs, but drop all palindromic SNPs</li>
</ol>
<p>By default, the <code>harmonise_data</code> function uses option 2, but this can be modified using the <code>action</code> argument, e.g. <code><a href="../reference/harmonise_data.html">harmonise_data(exposure_dat, outcome_dat, action=3)</a></code>.</p>
</div>
</div>
<div id="drop-duplicate-exposure-outcome-summary-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#drop-duplicate-exposure-outcome-summary-sets" class="anchor"></a>Drop duplicate exposure-outcome summary sets</h2>
<p>After data harmonisation, users may find that their dataset contains duplicate exposure-outcome summary sets. This can arise, for example, when a GWAS consortium has released multiple results from separate GWAS analyses for the same trait. For example, there are multiple GWAS summary datasets for body mass index and coronary heart disease:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ao</span><span class="op">[</span><span class="va">ao</span><span class="op">$</span><span class="va">trait</span><span class="op">==</span><span class="st">"Body mass index"</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"trait"</span>,<span class="st">"id"</span>,<span class="st">"pmid"</span>,<span class="st">"author"</span>,<span class="st">"sample_size"</span>,<span class="st">"nsnp"</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 15 x 6</span>
<span class="co">#&gt;    trait           id                   pmid author      sample_size     nsnp</span>
<span class="co">#&gt;    &lt;chr&gt;           &lt;chr&gt;               &lt;int&gt; &lt;chr&gt;             &lt;int&gt;    &lt;int&gt;</span>
<span class="co">#&gt;  1 Body mass index ieu-a-974        25673413 Locke AE         171977  2494613</span>
<span class="co">#&gt;  2 Body mass index ebi-a-GCST006802 26961502 Wood AR          119688  8580466</span>
<span class="co">#&gt;  3 Body mass index ieu-a-785        25673413 Locke AE         152893  2477659</span>
<span class="co">#&gt;  4 Body mass index ebi-a-GCST008025 31217584 Wojcik GL         21955 34343880</span>
<span class="co">#&gt;  5 Body mass index ieu-a-2          25673413 Locke AE         339224  2555511</span>
<span class="co">#&gt;  6 Body mass index ieu-a-1089       26961502 Wood             120286  8654252</span>
<span class="co">#&gt;  7 Body mass index bbj-a-1          28892062 Ishigaki K       158284  5961600</span>
<span class="co">#&gt;  8 Body mass index ieu-a-835        25673413 Locke AE         322154  2554668</span>
<span class="co">#&gt;  9 Body mass index ebi-a-GCST002783 25673413 Locke AE         236781  2529499</span>
<span class="co">#&gt; 10 Body mass index ieu-a-94         23754948 Randall JC        60586  2736876</span>
<span class="co">#&gt; 11 Body mass index ebi-a-GCST006368 30108127 Hoffmann TJ      315347 27854527</span>
<span class="co">#&gt; 12 Body mass index bbj-a-2          28892062 Ishigaki K        85894  6108953</span>
<span class="co">#&gt; 13 Body mass index bbj-a-3          28892062 Ishigaki K        72390  6108953</span>
<span class="co">#&gt; 14 Body mass index ebi-a-GCST004904 28892062 Akiyama M        158284  5952516</span>
<span class="co">#&gt; 15 Body mass index ieu-a-95         23754948 Randall JC        73137  2736876</span>
<span class="va">ao</span><span class="op">[</span><span class="va">ao</span><span class="op">$</span><span class="va">trait</span><span class="op">==</span><span class="st">"Coronary heart disease"</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"trait"</span>,<span class="st">"id"</span>,<span class="st">"pmid"</span>,<span class="st">"author"</span>,<span class="st">"ncase"</span>,<span class="st">"ncontrol"</span>,<span class="st">"nsnp"</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 5 x 7</span>
<span class="co">#&gt;   trait                id                  pmid author     ncase ncontrol   nsnp</span>
<span class="co">#&gt;   &lt;chr&gt;                &lt;chr&gt;              &lt;int&gt; &lt;chr&gt;      &lt;int&gt;    &lt;int&gt;  &lt;int&gt;</span>
<span class="co">#&gt; 1 Coronary heart dise~ ieu-a-6         21378988 Peden      15420    15062 5.40e5</span>
<span class="co">#&gt; 2 Coronary heart dise~ ieu-a-7         26343387 Nikpay     60801   123504 9.46e6</span>
<span class="co">#&gt; 3 Coronary heart dise~ ieu-a-8         21378990 Schunkert~ 22233    64762 2.42e6</span>
<span class="co">#&gt; 4 Coronary heart dise~ ebi-a-GCST0009~ 21378990 Schunkert~ 22233    64762 2.42e6</span>
<span class="co">#&gt; 5 Coronary heart dise~ ieu-a-9         23202125 Deloukas   63746   130681 7.91e4</span></code></pre></div>
<p>There are therefore multiple potential combinations of body mass index and coronary heart disease, which would likely lead to duplicate MR analyses. We recommend that users prune their datasets so that only the exposure-outcome combination with the highested expected power is retained. This can be done by selecting the exposure-outcome summary set with the largest sample size for the outcome, using the power_prune function:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/power_prune.html">power_prune</a></span><span class="op">(</span><span class="va">dat</span>,method<span class="op">=</span><span class="fl">1</span>,dist.outcome<span class="op">=</span><span class="st">"binary"</span><span class="op">)</span></code></pre></div>
<p>This drops the duplicate exposure-outcome sets with the smaller outcome sample size (number of cases for binary outcomes). Remaining duplicates are then dropped on the basis of the exposure sample size. However, if there are a large number of SNPs available to instrument an exposure, the outcome GWAS with the better SNP coverage may provide better power than the outcome GWAS with the larger sample size. This can occur, for example, if the larger outcome GWAS has used a targeted genotyping array. In such instances, it may be better to prune studies on the basis of instrument strength (i.e. variation in exposure explained by the instrumental SNPs) as well as sample size. This can be done by setting the method argument to 2:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/power_prune.html">power_prune</a></span><span class="op">(</span><span class="va">dat</span>,method<span class="op">=</span><span class="fl">2</span>,dist.outcome<span class="op">=</span><span class="st">"binary"</span><span class="op">)</span></code></pre></div>
<p>This procedure drops duplicate exposure-outcome sets on the basis of instrument strength and sample size, and assumes that the SNP-exposure effects correspond to a continuous trait with a normal distribution (i.e. exposure should not be binary). The SNP-outcome effects can correspond to either a binary or continuous trait (default behaviour is to assume a binary distribution). If the exposure is binary then method 1 should be used.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
