<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Outcome data • TwoSampleMR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Outcome data">
<meta property="og:description" content="TwoSampleMR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mrcieu/TwoSampleMR">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="outcome_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Outcome data</h1>
                        <h4 class="author">Gibran Hemani</h4>
            
      
      
      <div class="hidden name"><code>outcome.Rmd</code></div>

    </div>

    
    
<p>Once instruments for the exposure trait have been specified, those variants need to be extracted from the outcome trait.</p>
<div id="available-studies-in-ieu-gwas-database" class="section level2">
<h2 class="hasAnchor">
<a href="#available-studies-in-ieu-gwas-database" class="anchor"></a>Available studies in IEU GWAS database</h2>
<p>The IEU GWAS database (IGD) contains complete GWAS summary statistics from a large number of studies. You can browse them here:</p>
<p><a href="https://gwas.mrcieu.ac.uk/" class="uri">https://gwas.mrcieu.ac.uk/</a></p>
<p>To obtain details about the available GWASs programmatically do the following:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_outcomes.html">available_outcomes</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ao</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 22</span>
<span class="co">#&gt;   id      trait    note  group_name  year    mr author consortium sex   priority</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;      &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;    &lt;int&gt;</span>
<span class="co">#&gt; 1 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 2 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 3 ukb-e-~ Symptom~ NA    public      2020     1 Pan-U~ NA         Male~        0</span>
<span class="co">#&gt; 4 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 5 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; 6 eqtl-a~ ENSG000~ NA    public      2018     1 Vosa U NA         Male~        0</span>
<span class="co">#&gt; # ... with 12 more variables: population &lt;chr&gt;, unit &lt;chr&gt;, sample_size &lt;int&gt;,</span>
<span class="co">#&gt; #   nsnp &lt;int&gt;, build &lt;chr&gt;, subcategory &lt;chr&gt;, category &lt;chr&gt;, ontology &lt;chr&gt;,</span>
<span class="co">#&gt; #   ncase &lt;int&gt;, ncontrol &lt;int&gt;, pmid &lt;int&gt;, sd &lt;dbl&gt;</span></code></pre></div>
<p>For information about authentication see <a href="https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication">https://github.com/MRCIEU/ieugwasr/blob/master/README.md#authentication</a>.</p>
<p>The <code>available_outcomes</code> function returns a table of all the available studies in the database. Each study has a unique ID. e.g.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">ao</span>, select<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">trait</span>, <span class="va">id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   trait                               id                    </span>
<span class="co">#&gt;   &lt;chr&gt;                               &lt;chr&gt;                 </span>
<span class="co">#&gt; 1 ENSG00000198715                     eqtl-a-ENSG00000198715</span>
<span class="co">#&gt; 2 ENSG00000179950                     eqtl-a-ENSG00000179950</span>
<span class="co">#&gt; 3 Symptoms involving digestive system ukb-e-561_CSA         </span>
<span class="co">#&gt; 4 ENSG00000106683                     eqtl-a-ENSG00000106683</span>
<span class="co">#&gt; 5 ENSG00000143819                     eqtl-a-ENSG00000143819</span>
<span class="co">#&gt; 6 ENSG00000139574                     eqtl-a-ENSG00000139574</span></code></pre></div>
</div>
<div id="extracting-particular-snps-from-particular-studies" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-particular-snps-from-particular-studies" class="anchor"></a>Extracting particular SNPs from particular studies</h2>
<p>If we want to perform MR of BMI against coronary heart disease, we need to identify the SNPs that influence the BMI, and then extract those SNPs from a GWAS on coronary heart disease.</p>
<p>Let’s get the Locke et al 2014 instruments for BMI as an example:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bmi_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_instruments.html">extract_instruments</a></span><span class="op">(</span>outcomes<span class="op">=</span><span class="st">'ieu-a-2'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">bmi_exp_dat</span><span class="op">)</span>
<span class="co">#&gt;   pval.exposure samplesize.exposure chr.exposure se.exposure beta.exposure</span>
<span class="co">#&gt; 1   2.18198e-08              339152            1      0.0030       -0.0168</span>
<span class="co">#&gt; 2   4.56773e-11              339065            1      0.0031        0.0201</span>
<span class="co">#&gt; 3   5.05941e-14              313621            1      0.0087        0.0659</span>
<span class="co">#&gt; 4   5.45205e-10              338768            1      0.0029        0.0181</span>
<span class="co">#&gt; 5   1.88018e-28              338123            1      0.0030        0.0331</span>
<span class="co">#&gt; 6   2.28718e-40              339078            1      0.0037        0.0497</span>
<span class="co">#&gt;   pos.exposure id.exposure        SNP effect_allele.exposure</span>
<span class="co">#&gt; 1     47684677     ieu-a-2   rs977747                      G</span>
<span class="co">#&gt; 2     78048331     ieu-a-2 rs17381664                      C</span>
<span class="co">#&gt; 3    110082886     ieu-a-2  rs7550711                      T</span>
<span class="co">#&gt; 4    201784287     ieu-a-2  rs2820292                      C</span>
<span class="co">#&gt; 5     72837239     ieu-a-2  rs7531118                      C</span>
<span class="co">#&gt; 6    177889480     ieu-a-2   rs543874                      G</span>
<span class="co">#&gt;   other_allele.exposure eaf.exposure                      exposure</span>
<span class="co">#&gt; 1                     T       0.5333 Body mass index || id:ieu-a-2</span>
<span class="co">#&gt; 2                     T       0.4250 Body mass index || id:ieu-a-2</span>
<span class="co">#&gt; 3                     C       0.0339 Body mass index || id:ieu-a-2</span>
<span class="co">#&gt; 4                     A       0.5083 Body mass index || id:ieu-a-2</span>
<span class="co">#&gt; 5                     T       0.6083 Body mass index || id:ieu-a-2</span>
<span class="co">#&gt; 6                     A       0.2667 Body mass index || id:ieu-a-2</span>
<span class="co">#&gt;   mr_keep.exposure pval_origin.exposure data_source.exposure</span>
<span class="co">#&gt; 1             TRUE             reported                  igd</span>
<span class="co">#&gt; 2             TRUE             reported                  igd</span>
<span class="co">#&gt; 3             TRUE             reported                  igd</span>
<span class="co">#&gt; 4             TRUE             reported                  igd</span>
<span class="co">#&gt; 5             TRUE             reported                  igd</span>
<span class="co">#&gt; 6             TRUE             reported                  igd</span></code></pre></div>
<p>We now need to find a suitable GWAS for coronary heart disease. We can search the available studies:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ao</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"heart disease"</span>, <span class="va">ao</span><span class="op">$</span><span class="va">trait</span><span class="op">)</span>, <span class="op">]</span>
<span class="co">#&gt; # A tibble: 28 x 22</span>
<span class="co">#&gt;    id     trait   note   group_name  year    mr author consortium sex   priority</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;      &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;    &lt;int&gt;</span>
<span class="co">#&gt;  1 finn-~ Other ~ NA     public      2020     1 NA     NA         Male~        0</span>
<span class="co">#&gt;  2 ukb-b~ Diagno~ 41204~ public      2018     1 Ben E~ MRC-IEU    Male~        1</span>
<span class="co">#&gt;  3 ieu-a~ Corona~ &lt;NA&gt;   public      2011     1 Peden  C4D        Male~        3</span>
<span class="co">#&gt;  4 finn-~ Other ~ NA     public      2020     1 NA     NA         Male~        0</span>
<span class="co">#&gt;  5 ukb-b~ Diagno~ 41204~ public      2018     1 Ben E~ MRC-IEU    Male~        1</span>
<span class="co">#&gt;  6 finn-~ Major ~ NA     public      2020     1 NA     NA         Male~        0</span>
<span class="co">#&gt;  7 ukb-b~ Diagno~ 41204~ public      2018     1 Ben E~ MRC-IEU    Male~        1</span>
<span class="co">#&gt;  8 finn-~ Rheuma~ NA     public      2020     1 NA     NA         Male~        0</span>
<span class="co">#&gt;  9 ukb-d~ Major ~ NA     public      2018     1 Neale~ NA         Male~        0</span>
<span class="co">#&gt; 10 ukb-d~ Major ~ NA     public      2018     1 Neale~ NA         Male~        0</span>
<span class="co">#&gt; # ... with 18 more rows, and 12 more variables: population &lt;chr&gt;, unit &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_size &lt;int&gt;, nsnp &lt;int&gt;, build &lt;chr&gt;, subcategory &lt;chr&gt;,</span>
<span class="co">#&gt; #   category &lt;chr&gt;, ontology &lt;chr&gt;, ncase &lt;int&gt;, ncontrol &lt;int&gt;, pmid &lt;int&gt;,</span>
<span class="co">#&gt; #   sd &lt;dbl&gt;</span></code></pre></div>
<p>The most recent CARDIOGRAM GWAS is ID number <code>ieu-a-7</code>. We can extract the BMI SNPs from this GWAS as follows:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">chd_out_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_outcome_data.html">extract_outcome_data</a></span><span class="op">(</span>
    snps <span class="op">=</span> <span class="va">bmi_exp_dat</span><span class="op">$</span><span class="va">SNP</span>,
    outcomes <span class="op">=</span> <span class="st">'ieu-a-7'</span>
<span class="op">)</span>
<span class="co">#&gt; Extracting data for 79 SNP(s) from 1 GWAS(s)</span></code></pre></div>
<p>The <code>extract_outcome_data</code> is a flexible function. The <code>snps</code> argument only requires an array of rsIDs, and the <code>outcomes</code> variable can be a vector of outcomes. e.g. <code>chd_out_dat(c("rs234", "rs17097147"), c('ieu-a-2', 'ieu-a-7'))</code> will extract the two SNPs from each of the outcomes <code>ieu-a-2</code> and <code>ieu-a-7</code>.</p>
</div>
<div id="ld-proxies" class="section level2">
<h2 class="hasAnchor">
<a href="#ld-proxies" class="anchor"></a>LD proxies</h2>
<p>By default if a particular requested SNP is not present in the outcome GWAS then a SNP (proxy) that is in LD with the requested SNP (target) will be searched for instead. LD proxies are defined using 1000 genomes European sample data. The effect of the proxy SNP on the outcome is returned, along with the proxy SNP, the effect allele of the proxy SNP, and the corresponding allele (in phase) for the target SNP.</p>
<p>The parameters for handling LD proxies are as follows:</p>
<ul>
<li>
<code>proxies</code> = TRUE or FALSE (TRUE by default)</li>
<li>
<code>rsq</code> = numeric value of minimum rsq to find a proxy. Default is 0.8, minimum is 0.6</li>
<li>
<code>palindromes</code> = Allow palindromic SNPs? Default is 1 (yes)</li>
<li>
<code>maf_threshold</code> = If palindromes allowed then what is the maximum minor allele frequency of palindromes allowed? Default is 0.3.</li>
</ul>
</div>
<div id="using-local-gwas-summary-data" class="section level2">
<h2 class="hasAnchor">
<a href="#using-local-gwas-summary-data" class="anchor"></a>Using local GWAS summary data</h2>
<p>If you have GWAS summary data that is not present in IEU GWAS database, this can still be used to perform analysis.</p>
<p>Supposing there is a GWAS summary file called “gwas_summary.csv” with e.g. 2 million rows and it looks like this:</p>
<pre><code>rsid,effect,SE,a1,a2,a1_freq,p-value,Units,Gene,n
rs10767664,0.19,0.030612245,A,T,0.78,5.00E-26,kg/m2,BDNF,225238
rs13078807,0.1,0.020408163,G,A,0.2,4.00E-11,kg/m2,CADM2,221431
rs1514175,0.07,0.020408163,A,G,0.43,8.00E-14,kg/m2,TNNI3K,207641
rs1558902,0.39,0.020408163,A,T,0.42,5.00E-120,kg/m2,FTO,222476
...
...</code></pre>
<p>To extract the exposure SNPs from this data, we would use the following command:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outcome_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_outcome_data.html">read_outcome_data</a></span><span class="op">(</span>
    snps <span class="op">=</span> <span class="va">bmi_exp_dat</span><span class="op">$</span><span class="va">SNP</span>,
    filename <span class="op">=</span> <span class="st">"gwas_summary.csv"</span>,
    sep <span class="op">=</span> <span class="st">","</span>,
    snp_col <span class="op">=</span> <span class="st">"rsid"</span>,
    beta_col <span class="op">=</span> <span class="st">"effect"</span>,
    se_col <span class="op">=</span> <span class="st">"SE"</span>,
    effect_allele_col <span class="op">=</span> <span class="st">"a1"</span>,
    other_allele_col <span class="op">=</span> <span class="st">"a2"</span>,
    eaf_col <span class="op">=</span> <span class="st">"a1_freq"</span>,
    pval_col <span class="op">=</span> <span class="st">"p-value"</span>,
    units_col <span class="op">=</span> <span class="st">"Units"</span>,
    gene_col <span class="op">=</span> <span class="st">"Gene"</span>,
    samplesize_col <span class="op">=</span> <span class="st">"n"</span>
<span class="op">)</span></code></pre></div>
<p>This returns an outcome data frame with only the SNPs that were requested (if those SNPs were present in the “gwas_summary.csv” file).</p>
</div>
<div id="outcome-data-format" class="section level2">
<h2 class="hasAnchor">
<a href="#outcome-data-format" class="anchor"></a>Outcome data format</h2>
<p>The <code>extract_outcome_data</code> function returns a table of SNP effects for the requested SNPs on the requested outcomes. The format of the data is similar to the exposure data format, except the main columns are as follows:</p>
<ul>
<li><code>SNP</code></li>
<li><code>beta.outcome</code></li>
<li><code>se.outcome</code></li>
<li><code>samplesize.outcome</code></li>
<li><code>ncase.outcome</code></li>
<li><code>ncontrol.outcome</code></li>
<li><code>pval.outcome</code></li>
<li><code>eaf.outcome</code></li>
<li><code>effect_allele.outcom</code></li>
<li><code>other_allele.outcome</code></li>
<li><code>units.outcome</code></li>
<li><code>outcome</code></li>
<li><code>consortium.outcome</code></li>
<li><code>year.outcome</code></li>
<li><code>pmid.outcome</code></li>
<li><code>id.outcome</code></li>
<li><code>originalname.outcome</code></li>
<li><code>proxy.outcome</code></li>
<li><code>target_snp.outcome</code></li>
<li><code>proxy_snp.outcome</code></li>
<li><code>target_a1.outcome</code></li>
<li><code>target_a2.outcome</code></li>
<li><code>proxy_a1.outcome</code></li>
<li><code>proxy_a2.outcome</code></li>
<li><code>mr_keep.outcome</code></li>
<li><code>data_source.outcome</code></li>
</ul>
</div>
<div id="more-advanced-use-of-local-data" class="section level2">
<h2 class="hasAnchor">
<a href="#more-advanced-use-of-local-data" class="anchor"></a>More advanced use of local data</h2>
<p>We have developed a summary data format called “GWAS VCF,” which is designed to store GWAS results in a strict and performant way. It is possible to use this format with the TwoSampleMR package. Going down this avenue also allows you to use LD proxy functionality using your own LD reference files (or ones that we provide). For more details, see this package that explains the format and how to query it in R:</p>
<p><a href="https://github.com/mrcieu/gwasvcf" class="uri">https://github.com/mrcieu/gwasvcf</a></p>
<p>and this package for how to connect the data to other packages including TwoSampleMR</p>
<p><a href="https://github.com/MRCIEU/gwasglue" class="uri">https://github.com/MRCIEU/gwasglue</a></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
