<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ TwoSampleMR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TwoSampleMR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/index.html">Guide</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mrcieu/TwoSampleMR">Source</a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="add_metadata.html">add_metadata()</a></code> </p>
        </td>
        <td><p>Add meta data to extracted data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="add_rsq.html">add_rsq()</a></code> </p>
        </td>
        <td><p>Estimate r-square of each association</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="allele_frequency.html">allele_frequency()</a></code> </p>
        </td>
        <td><p>Estimate allele frequency from SNP</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="available_outcomes.html">available_outcomes()</a></code> </p>
        </td>
        <td><p>Get list of studies with available GWAS summary statistics through API</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clump_data.html">clump_data()</a></code> </p>
        </td>
        <td><p>Perform LD clumping on SNP data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="combine_all_mrresults.html">combine_all_mrresults()</a></code> </p>
        </td>
        <td><p>Combine all mr results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="combine_data.html">combine_data()</a></code> </p>
        </td>
        <td><p>Combine data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="contingency.html">contingency()</a></code> </p>
        </td>
        <td><p>Obtain 2x2 contingency table from marginal parameters and odds ratio</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_outcome_to_exposure.html">convert_outcome_to_exposure()</a></code> </p>
        </td>
        <td><p>Convert outcome data to exposure data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dat_to_MRInput.html">dat_to_MRInput()</a></code> </p>
        </td>
        <td><p>Convert TwoSampleMR format to MendelianRandomization format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dat_to_RadialMR.html">dat_to_RadialMR()</a></code> </p>
        </td>
        <td><p>Convert dat to RadialMR format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="default_parameters.html">default_parameters()</a></code> </p>
        </td>
        <td><p>List of parameters for use with MR functions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="directionality_test.html">directionality_test()</a></code> </p>
        </td>
        <td><p>Perform MR Steiger test of directionality</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="effective_n.html">effective_n()</a></code> </p>
        </td>
        <td><p>Estimate the effective sample size in a case control study</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="enrichment.html">enrichment()</a></code> </p>
        </td>
        <td><p>Perform enrichment analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="enrichment_method_list.html">enrichment_method_list()</a></code> </p>
        </td>
        <td><p>Get list of available p-value enrichment methods</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="estimate_trait_sd.html">estimate_trait_sd()</a></code> </p>
        </td>
        <td><p>Estimate trait SD by obtaining beta estimates from z-scores and finding the ratio with original beta values</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extract_instruments.html">extract_instruments()</a></code> </p>
        </td>
        <td><p>Find instruments for use in MR from the MR Base database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extract_outcome_data.html">extract_outcome_data()</a></code> </p>
        </td>
        <td><p>Supply the output from <code>read_exposure_data</code> and all the SNPs therein will be queried against the requested outcomes in remote database using API.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fishers_combined_test.html">fishers_combined_test()</a></code> </p>
        </td>
        <td><p>Fisher's combined test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="forest_plot.html">forest_plot()</a></code> </p>
        </td>
        <td><p>Forest plot for multiple exposures and multiple outcomes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="forest_plot_1_to_many.html">forest_plot_1_to_many()</a></code> </p>
        </td>
        <td><p>1-to-many forest plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="forest_plot_basic2.html">forest_plot_basic2()</a></code> </p>
        </td>
        <td><p>A basic forest plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_1_to_many.html">format_1_to_many()</a></code> </p>
        </td>
        <td><p>Format MR results for a 1-to-many forest plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_aries_mqtl.html">format_aries_mqtl()</a></code> </p>
        </td>
        <td><p>Get data from methylation QTL results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_data.html">format_data()</a></code> </p>
        </td>
        <td><p>Read exposure or outcome data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_gtex_eqtl.html">format_gtex_eqtl()</a></code> </p>
        </td>
        <td><p>Get data from eQTL catalog into correct format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_gwas_catalog.html">format_gwas_catalog()</a></code> </p>
        </td>
        <td><p>Get data selected from GWAS catalog into correct format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_metab_qtls.html">format_metab_qtls()</a></code> </p>
        </td>
        <td><p>Get data from metabolomic QTL results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_mr_results.html">format_mr_results()</a></code> </p>
        </td>
        <td><p>Format MR results for forest plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="format_proteomic_qtls.html">format_proteomic_qtls()</a></code> </p>
        </td>
        <td><p>Get data from proteomic QTL results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate_odds_ratios.html">generate_odds_ratios()</a></code> </p>
        </td>
        <td><p>Generate odds ratios</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_population_allele_frequency.html">get_population_allele_frequency()</a></code> </p>
        </td>
        <td><p>Estimate the allele frequency in population from case/control summary data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_p_from_r2n.html">get_p_from_r2n()</a></code> </p>
        </td>
        <td><p>Calculate p-value from rsq and sample size</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_r_from_lor.html">get_r_from_lor()</a></code> </p>
        </td>
        <td><p>Estimate proportion of variance of liability explained by SNP in general population</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_r_from_pn.html">get_r_from_pn()</a></code> </p>
        </td>
        <td><p>Calculate variance explained from p vals and sample size</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_se.html">get_se()</a></code> </p>
        </td>
        <td><p>Get SE from effect size and pval</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="harmonise_data.html">harmonise_data()</a></code> </p>
        </td>
        <td><p>Harmonise the alleles and effects between the exposure and outcome</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="harmonise_ld_dat.html">harmonise_ld_dat()</a></code> </p>
        </td>
        <td><p>Harmonise LD matrix against summary data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Isq.html">Isq()</a></code> </p>
        </td>
        <td><p>I-square calculation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ldsc_h2.html">ldsc_h2()</a></code> </p>
        </td>
        <td><p>Univariate LDSC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ldsc_rg.html">ldsc_rg()</a></code> </p>
        </td>
        <td><p>Bivariate LDSC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ld_matrix.html">ld_matrix()</a></code> </p>
        </td>
        <td><p>Get LD matrix for list of SNPs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_dat.html">make_dat()</a></code> </p>
        </td>
        <td><p>Convenient function to create a harmonised dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr.html">mr()</a></code> </p>
        </td>
        <td><p>Perform all Mendelian randomization tests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_density_plot.html">mr_density_plot()</a></code> </p>
        </td>
        <td><p>Density plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_egger_regression.html">mr_egger_regression()</a></code> </p>
        </td>
        <td><p>Egger's regression for Mendelian randomization</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_egger_regression_bootstrap.html">mr_egger_regression_bootstrap()</a></code> </p>
        </td>
        <td><p>Run bootstrap to generate standard errors for MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_forest_plot.html">mr_forest_plot()</a></code> </p>
        </td>
        <td><p>Forest plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_funnel_plot.html">mr_funnel_plot()</a></code> </p>
        </td>
        <td><p>Funnel plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_heterogeneity.html">mr_heterogeneity()</a></code> </p>
        </td>
        <td><p>Get heterogeneity statistics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_ivw.html">mr_ivw()</a></code> </p>
        </td>
        <td><p>Inverse variance weighted regression</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_ivw_fe.html">mr_ivw_fe()</a></code> </p>
        </td>
        <td><p>Inverse variance weighted regression (fixed effects)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_ivw_mre.html">mr_ivw_mre()</a></code> </p>
        </td>
        <td><p>Inverse variance weighted regression (multiplicative random effects model)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_ivw_radial.html">mr_ivw_radial()</a></code> </p>
        </td>
        <td><p>Radial IVW analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_leaveoneout.html">mr_leaveoneout()</a></code> </p>
        </td>
        <td><p>Leave one out sensitivity analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_leaveoneout_plot.html">mr_leaveoneout_plot()</a></code> </p>
        </td>
        <td><p>Plot results from leaveoneout analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_median.html">mr_median()</a></code> </p>
        </td>
        <td><p>MR median estimators</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_meta_fixed.html">mr_meta_fixed()</a></code> </p>
        </td>
        <td><p>Perform 2 sample IV using fixed effects meta analysis and delta method for standard errors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_meta_fixed_simple.html">mr_meta_fixed_simple()</a></code> </p>
        </td>
        <td><p>Perform 2 sample IV using simple standard error</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_meta_random.html">mr_meta_random()</a></code> </p>
        </td>
        <td><p>Perform 2 sample IV using random effects meta analysis and delta method for standard errors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_method_list.html">mr_method_list()</a></code> </p>
        </td>
        <td><p>Get list of available MR methods</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_mode.html">mr_mode()</a></code> </p>
        </td>
        <td><p>MR mode estimators</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_moe.html">mr_moe()</a></code> </p>
        </td>
        <td><p>Mixture of experts</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_penalised_weighted_median.html">mr_penalised_weighted_median()</a></code> </p>
        </td>
        <td><p>Penalised weighted median MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_pleiotropy_test.html">mr_pleiotropy_test()</a></code> </p>
        </td>
        <td><p>Test for horizontal pleiotropy in MR analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_raps.html">mr_raps()</a></code> </p>
        </td>
        <td><p>Robust adjusted profile score</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_report.html">mr_report()</a></code> </p>
        </td>
        <td><p>Generate MR report</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_rucker.html">mr_rucker()</a></code> </p>
        </td>
        <td><p>MR Rucker framework</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_rucker_bootstrap.html">mr_rucker_bootstrap()</a></code> </p>
        </td>
        <td><p>Run rucker with bootstrap estimates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_rucker_cooksdistance.html">mr_rucker_cooksdistance()</a></code> </p>
        </td>
        <td><p>MR Rucker with outliers automatically detected and removed</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_rucker_jackknife.html">mr_rucker_jackknife()</a></code> </p>
        </td>
        <td><p>Run rucker with jackknife estimates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_scatter_plot.html">mr_scatter_plot()</a></code> </p>
        </td>
        <td><p>Create scatter plot with lines showing the causal estimate for different MR tests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_sign.html">mr_sign()</a></code> </p>
        </td>
        <td><p>MR sign test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_simple_median.html">mr_simple_median()</a></code> </p>
        </td>
        <td><p>Simple median method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_simple_mode.html">mr_simple_mode()</a></code> </p>
        </td>
        <td><p>MR simple mode estimator</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_simple_mode_nome.html">mr_simple_mode_nome()</a></code> </p>
        </td>
        <td><p>MR simple mode estimator (NOME)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_singlesnp.html">mr_singlesnp()</a></code> </p>
        </td>
        <td><p>Perform 2 sample MR on each SNP individually</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_steiger.html">mr_steiger()</a></code> </p>
        </td>
        <td><p>MR Steiger test of directionality</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_steiger2.html">mr_steiger2()</a></code> </p>
        </td>
        <td><p>MR Steiger test of directionality</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_two_sample_ml.html">mr_two_sample_ml()</a></code> </p>
        </td>
        <td><p>Maximum likelihood MR method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_uwr.html">mr_uwr()</a></code> </p>
        </td>
        <td><p>Unweighted regression</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_wald_ratio.html">mr_wald_ratio()</a></code> </p>
        </td>
        <td><p>Perform 2 sample IV using Wald ratio.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_weighted_median.html">mr_weighted_median()</a></code> </p>
        </td>
        <td><p>Weighted median method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_weighted_mode.html">mr_weighted_mode()</a></code> </p>
        </td>
        <td><p>MR weighted mode estimator</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_weighted_mode_nome.html">mr_weighted_mode_nome()</a></code> </p>
        </td>
        <td><p>MR weighted mode estimator (NOME)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mr_wrapper.html">mr_wrapper()</a></code> </p>
        </td>
        <td><p>Perform full set of MR analyses</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_basic.html">mv_basic()</a></code> </p>
        </td>
        <td><p>Perform basic multivariable MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_extract_exposures.html">mv_extract_exposures()</a></code> </p>
        </td>
        <td><p>Extract exposure variables for multivariable MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_extract_exposures_local.html">mv_extract_exposures_local()</a></code> </p>
        </td>
        <td><p>Attempt to perform MVMR using local data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_harmonise_data.html">mv_harmonise_data()</a></code> </p>
        </td>
        <td><p>Harmonise exposure and outcome for multivariable MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_ivw.html">mv_ivw()</a></code> </p>
        </td>
        <td><p>Perform IVW multivariable MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_lasso_feature_selection.html">mv_lasso_feature_selection()</a></code> </p>
        </td>
        <td><p>Apply LASSO feature selection to mvdat object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_multiple.html">mv_multiple()</a></code> </p>
        </td>
        <td><p>Perform IVW multivariable MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_residual.html">mv_residual()</a></code> </p>
        </td>
        <td><p>Perform basic multivariable MR</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mv_subset.html">mv_subset()</a></code> </p>
        </td>
        <td><p>Perform multivariable MR on subset of features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="power_prune.html">power_prune()</a></code> </p>
        </td>
        <td><p>Power prune</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_exposure_data.html">read_exposure_data()</a></code> </p>
        </td>
        <td><p>Read exposure data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_outcome_data.html">read_outcome_data()</a></code> </p>
        </td>
        <td><p>Read outcome data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="run_mrmix.html">run_mrmix()</a></code> </p>
        </td>
        <td><p>Perform MRMix analysis on harmonised dat object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="run_mr_presso.html">run_mr_presso()</a></code> </p>
        </td>
        <td><p>Wrapper for MR-PRESSO</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="size.prune.html">size.prune()</a></code> </p>
        </td>
        <td><p>Size prune</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sort_1_to_many.html">sort_1_to_many()</a></code> </p>
        </td>
        <td><p>Sort results for 1-to-many forest plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="split_exposure.html">split_exposure()</a></code> </p>
        </td>
        <td><p>Split exposure column</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="split_outcome.html">split_outcome()</a></code> </p>
        </td>
        <td><p>Split outcome column</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="standardise_units.html">standardise_units()</a></code> </p>
        </td>
        <td><p>Try to standardise continuous traits to be in standard deviation units</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="steiger_filtering.html">steiger_filtering()</a></code> </p>
        </td>
        <td><p>Steiger filtering function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="steiger_sensitivity.html">steiger_sensitivity()</a></code> </p>
        </td>
        <td><p>Evaluate the Steiger test's sensitivity to measurement error</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subset_on_method.html">subset_on_method()</a></code> </p>
        </td>
        <td><p>Subset MR-results on method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trim.html">trim()</a></code> </p>
        </td>
        <td><p>Trim function to remove leading and trailing blank spaces</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="TwoSampleMR-package.html">TwoSampleMR-package</a></code> </p>
        </td>
        <td><p>TwoSampleMR: Two Sample MR functions and interface to MR Base database</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="weighted_median.html">weighted_median()</a></code> </p>
        </td>
        <td><p>Weighted median method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="weighted_median_bootstrap.html">weighted_median_bootstrap()</a></code> </p>
        </td>
        <td><p>Calculate standard errors for weighted median method using bootstrap</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Gibran Hemani, Philip Haycock, Jie Zheng, Tom Gaunt, Ben Elsworth, Tom Palmer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


